### PR TITLE
[AWS][Lambda] Add dimension fields for TSDB support

### PR DESCRIPTION
**Pull request by constanca-m**
## What does this PR do?

Add dimension fields to Lambda datastream.

## Details

The lambda function name is unique per region within an account. Since the lambda function can have different versions, we need to make that a dimension as well. As the version affects the resource name, this will be our last dimension.

There was not a change in the number of documents after enabling TSDB:

image

Tested with one account, one lambda function with two different versions.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## How to test this PR locally

Refer to https://github.com/elastic/integrations/issues/5970

## Related issues

- Relates to https://github.com/elastic/integrations/issues/5970

### CODE REVIEW

File: packages/aws/data_stream/lambda/fields/fields.yml [hunk 1]
```diff
@@ -6,12 +6,15 @@
       fields:
         - name: FunctionName
           type: keyword
+          dimension: true
           description: Lambda function name.
         - name: Resource
           type: keyword
+          dimension: true
```
Comment (tetianakravchenko) [inline packages/aws/data_stream/lambda/fields/fields.yml:13]: what is a meaning of the `Resource` field? From the issue description I got an impression that name of function and version uniquely identify the function

Comment (constanca-m) [reply packages/aws/data_stream/lambda/fields/fields.yml:13]: image

Without setting the Resource as dimension, the first 2 documents will be overwriting each other @tetianakravchenko

Comment (constanca-m) [reply packages/aws/data_stream/lambda/fields/fields.yml:13]: For example, without setting `Resource` as dimension, these 2 documents will be overwriting each other:
image

@tetianakravchenko

Comment (tetianakravchenko) [reply packages/aws/data_stream/lambda/fields/fields.yml:13]: what is the difference between those documents?

Comment (constanca-m) [reply packages/aws/data_stream/lambda/fields/fields.yml:13]: Good question. The only difference from comparing these two documents is than one has the dimension resource and the other has not, but even the values for the metrics are exactly the same @tetianakravchenko I am not sure if I should let be overwritten then or how to make sure that the one with the `resource` set will be the final document.

Comment (tetianakravchenko) [reply packages/aws/data_stream/lambda/fields/fields.yml:13]: @elastic/obs-cloud-monitoring is it an expected behavior that there are 2 documents with the same data?

@constanca-m so far it seems for me that generic pattern for aws package datasreams is to add all fields that are part of `dimentions.*` object

**Review by tetianakravchenko — APPROVED** [review]

**Review by kaiyan-sheng — APPROVED** [review]

**Pull request merged**